test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -529,13 +532,28 @@ public void testToQuery() throws IOException {
         }
     }
 
+    // TODO: Deprecate this method?
     protected QueryBuilder rewriteQuery(QB queryBuilder, QueryRewriteContext rewriteContext) throws IOException {
         QueryBuilder rewritten = rewriteAndFetch(queryBuilder, rewriteContext);
         // extra safety to fail fast - serialize the rewritten version to ensure it's serializable.
         assertSerialization(rewritten);
         return rewritten;
     }
 
+    protected QueryBuilder rewriteQuery(
+        QB queryBuilder,
+        QueryRewriteContext coordinatorRewriteContext,
+        SearchExecutionContext shardRewriteContext
+    ) throws IOException {
+        // The first rewriteAndFetch call simulates rewriting on the coordinator node
+        // The second rewriteAndFetch call simulates rewriting on the shard
+        QueryBuilder rewritten = rewriteAndFetch(queryBuilder, coordinatorRewriteContext);
```

Comment:
Question: should we call `assertSerialization` after this? I imagine the query is transferred to the shard by being serialized...